Protocol.md

```diff
@@ -0,0 +1,187 @@
+The following figure gives a brief overview of the protocol:
+
+    +-------+                    +-------+
+    |       |                    |       |
+    |       |---- (1) offer  --->|       |
+    |       |                    |       |
+    |       |<--- (2) accept ----|       |
+    |   A   |                    |   B   |
+    |       |---- (3) sign   --->|       |
+    |       |                    |       |
+    |       |                    |  (4) broadcast fund-tx
+    |       |                    |       |
+    +-------+                    +-------+
+
+### Offer
+
+The initiating party starts the protocol by sending an `offer` message to the other party.
+This message includes the following information:
+1. Contract information (an array of triples defined below)
+1. Oracle information (optional)
+1. A's fund amount (satoshi)
+1. A's [extended public key](https://github.com/discreetlogcontracts/dlcspecs/blob/master/KeyDerivation.md)
+1. A's inputs
+1. A's change address
+1. Estimatesmartfee (satoshi/vbyte)
+1. CET CSV delay
+1. CET locktime (corresponding to the contract maturity date)
+1. Refund locktime
+
+#### Contract information
+Contract information consists of a total value (sum of both outputs) and a map to be used to create CETs.
+The map consists of:
+`message` - `a_output` (satoshi)
+where `message` is the data on which the oracle will be providing a (set of) signatures, and `a_output` represents the amount that A will receive for this outcome (B's outcome can be computed by substracting `a_output` to the total value).
+The interpretation of the message semantics is left to the application.
+
+#### Oracle information
+The oracle public key and R point(s) to use to build the CETs as well as meta information to identify the oracle to be used in the contract.
+Example:
+`public_key` - `R_point`<sub>1</sub> - `R_point`<sub>2</sub> - `meta data`
+The meta data can either be human readable for a user to be able to interpret it, or machine readable so that an application can interpret it and present it to the user.
+If both parties already have this information, transmission is unnecessary.
+
+#### A's fund amount (satoshi)
+How much A inputs into the contract.
+
+#### A's extended public key
+The extended public key to be used to [derive](https://github.com/discreetlogcontracts/dlcspecs/blob/master/KeyDerivation.md) public keys for the Fund transaction, refund transaction and CETs.
+
+#### A's inputs
+The set of UTXOs to be used as input to the fund transaction.
+
+#### A's change address
+The address to use to send the change if the sum of A's inputs is greater than the fund amount plus the fees.
+
+#### Estimatesmartfee (satoshi/vbyte)
+The fee rate to be used when computing fees for the different transactions.
+
+#### CET CSV Delay
+The number to use as input to the CSV in the CET transactions.
```

Comment:
nit: link to the `ToLocalOutput` of `Transacitons.md` and also the T in CET is transaction so probably just CETs or spelling it all out will do

Reply:
Added a link to the output section of CET in `Transaction.md` (that's the closest thing I could link to I think)